test/built-ins/TypedArrays/of/argument-number-value-throws.js

```diff
@@ -0,0 +1,33 @@
+// Copyright (C) 2016 the V8 project authors. All rights reserved.
+// This code is governed by the BSD license found in the LICENSE file.
+/*---
+id: sec-%typedarray%.from
+description: >
+  Return abrupt from object value
+info: >
+  22.2.2.2 %TypedArray%.of ( ...items )
+
+  ...
+  7. Repeat, while k < len
+    ...
+    c. Perform ? Set(newObj, Pk, kValue, true).
+  ...
+includes: [testTypedArray.js]
+---*/
+
+var obj1 = {
+  valueOf() {
+    return 42;
+  }
+};
+var obj2 = {
+  valueOf() {
+    throw new Test262Error();
+  }
+};
+
+testWithTypedArrayConstructors(function(TA) {
+  assert.throws(Test262Error, function() {
+    TA.of(obj1, obj2);
```

Comment:
You might use the same pattern from `set-value-abrupt-completion.js` to ensure correct iteration cancelling: have each setter close around a `lastValue` binding, and update the invocation to read: `TA.of(obj1, obj2, obj1);`